### ISSUE TITLE
Handle project access for non-repository members

### ISSUE THREAD

**Issue by gtsiolis** [gitpod-io/gitpod#7159]
### Problem to solve

Users that don't have access to a repository they run into errors and empty pages when access a project in Gitpod.

### Proposal

TBD (To be discussed)

This can become more relevant as users adopt **Teams & Projects** and create teams with multiple projects across different providers with team members that don't necessarily have access to corresponding repositories. 💡

**Comment by jldec** [comment]
Hi @gtsiolis would you mind please adding the steps required to reproduce this issue.

**Comment by gtsiolis** [comment]
@jldec I've added steps to reproduce in the issue description to clarify. Let me know if these don't look clear enough.

**Comment by jldec** [comment]
Thanks - I suppose we could always improve the error message, but the behavior seems correct - users should not be able to see repositories through Gitpod which they cannot acces directly.

**Comment by gtsiolis** [comment]
> I suppose we could always improve the error message, but the behavior seems correct 

Yes, handling this with better error messages is a good way forward. since there's no much we can do about this and we most probably don't want to limit project visibility for users that are part of a team but don't have access to the actual repository. 🧀

**Comment by jldec** [comment]
Looking at the error message which is there, that seems pretty close to what we want for workspace startup.
It is consistent with GitHub's behavior of showing 404 for private repos accessed by unauthed users.
We don't want to leak more information to unauthorized users.

What might be more promising as a way to improve the UX would be to show projects which won't work for you because of auth-failures in the project list, and team members with such project auth issues in the team members list.
